### ISSUE TITLE
Different types with the same name should cause `Duplicate typename` error

### ISSUE THREAD

**Issue by hit9** [eleme/thriftpy#76]
E.g. the following thrift should cause `duplicate typename` error

```
struct foo {}
enum foo {}
```
But this wasn't handled in thriftpy.

**Comment by hit9** [comment]
excludes `namespace`, all other types: `const`, `enum`, `struct`, `union`, `exception`, `service`, should use the same namespace, that means they cannot use the same name with different types.

**Comment by hit9** [comment]
New thrift parser model `thrift` should contains:

```
includes
constants
namespaces
ttypes
```

**Comment by hit9** [comment]
Also the field names should be independent in a srruct/function.

**Comment by hit9** [comment]
And deprecate union support (What data structure in py should I use for union..

**Comment by hit9** [comment]
Will be fixed in #80

**Comment by hit9** [comment]
Closed via #80 .
